### PR TITLE
Fix: Add all challenges for first session.

### PR DESCRIPTION
**Pull request by gabrielmocanu**
Signed-off-by: Gabriel Mocanu <gabi.mocanu98@gmail.com>

### CODE REVIEW

**Comment by gabrielmocanu** [comment]
@carrrina  Can you take a look at this PR?

File: web-basics-browser-security-model/README.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+
+- **Addressing:** resources on the web are located using the URI/URL addressing scheme. Possible vulnerabilities here include a misconfigured web server that allows viewing application-specific files, or worse, that allows accessing other files on the host machine. While this information leakage is not very dangerous by itself, it may be used as an intermediary stage for other attacks. You can read more about URLs here [[1]](https://skorks.com/2010/05/what-every-developer-should-know-about-urls/).
+
+- **Request methods:** HTTP communication is done by using methods, also called HTTP verbs. The most used methods are GET, POST, PUT and DELETE. GET method is read-only and is used to retrieve data from the server. DELETE request is used to remove the specified resource from the server. PUT method is used to modify an entire resource. POST requests are used to create new resources. You can find more information about all methods here [[2]](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods).
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:17]:
```suggestion
- **Request methods:** HTTP communication is done by using methods, also called HTTP verbs. The most used methods are GET, POST, PUT and DELETE. GET method is read-only and is used to retrieve data from the server. DELETE request is used to remove the specified resource from the server. PUT method is used to modify an entire resource. POST requests are used to create new resources. You can find more information about all methods here [[2]](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods).
```
```suggestion
- **Request methods:** HTTP communication is done by using methods, also called HTTP verbs. The most used methods are GET, POST, PUT and DELETE. The GET method is read-only and is used to retrieve data from the server. A DELETE request is used to remove the specified resource from the server. The PUT method is used to modify an entire resource. POST requests are used to create new resources. You can find more information about all methods here [[2]](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods).
```

File: web-basics-browser-security-model/README.md [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+
+A typical HTTP request / response circle:
+
+1. The browser requests an HTML page. The server returns an HTML file.
+2. The browser requests a style sheet. The server returns a CSS file.
+3. The browser requests a JPEG image. The server returns a JPG file.
+4. The browser requests JavaScript code. The server returns a JS file
+5. The browser requests data. The server returns data (in XML or JSON).
+
+### XHR - XML Http Request
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:84]:
```suggestion
### XHR - XML Http Request
```
```suggestion
### XHR - XMLHttpRequest
```

Comment (carrrina) [inline web-basics-browser-security-model/README.md:84]:
```suggestion
### XHR - XML Http Request
```
```suggestion
### XHR - XMLHttpRequest
```

File: web-basics-browser-security-model/README.md [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+  Also, the [DOM](https://github.com/hexcellents/sss-web/wiki/Session-01:-Web-Basics-&-Browser-Security-Model#dom-document-object-model) (Document Object Model) structure of the page can be altered, and elements can be added or removed, such as scripts, input fields, etc. (any element in fact), which means that any JavaScript code used to sanitize user input or perform other functions can be bypassed.
+
+![Elements - Developer Tools](./assets/devtools-1.png)
+
+- **Console**: The console prints errors which occurred during page rendering or during any action performed on the page, such as, but not limited to, error loading an image not found, error while performing an asynchronous request to fetch data, missing included file (such as CSS or Javascript files), errors in Javascript code from the included scripts, debug messages left by the developer, etc.
+  The console also has the ability to run any Javascript code by typing it directly there and interacting with the page.
+
+![Console - Developer Tools](./assets/devtools-2.png)
+
+- **Sources**: This tab lets you see any file from the loaded in the front-end, such as images, JS, CSS, etc. in an arborescent way.
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:179]:
```suggestion
- **Sources**: This tab lets you see any file from the loaded in the front-end, such as images, JS, CSS, etc. in an arborescent way.
```
```suggestion
- **Sources**: This tab lets you see any file loaded in the front-end, such as images, JS, CSS, etc. in an arborescent way.
```

Comment (carrrina) [inline web-basics-browser-security-model/README.md:179]:
```suggestion
- **Sources**: This tab lets you see any file from the loaded in the front-end, such as images, JS, CSS, etc. in an arborescent way.
```
```suggestion
- **Sources**: This tab lets you see any file loaded in the front-end, such as images, JS, CSS, etc. in an arborescent way.
```

File: web-basics-browser-security-model/README.md [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+Windows may contain frames from different sources
+
+- Frame: rigid division as part of frameset
+- iFrame: floating inline frame
+
+iFrame example:
+
+```
+<iframe src="simple_iframe.html" width=450 height=100>
+if you can see this, you browser doesn't understand IFRAME.
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:265]:
```suggestion
if you can see this, you browser doesn't understand IFRAME.
```
```suggestion
if you can see this, your browser doesn't understand IFRAME.
```

Comment (carrrina) [inline web-basics-browser-security-model/README.md:265]:
```suggestion
if you can see this, you browser doesn't understand IFRAME.
```
```suggestion
if you can see this, your browser doesn't understand IFRAME.
```

File: web-basics-browser-security-model/README.md [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+- prevent links from targeting other browsing contexts
+- prevent content from using plugins (through \<embed\>, \<object\>, \<applet\> or others)
+- prevent the content to navigate its top-level browsing context
+- block automatically triggered features (such as automatically playing a video or automatically focusing a form control)
+
+Add the below HTML code in the same index.html as above:
+
+```
+<iframe src="sandbox_iframe.html" sandbox width=450 height=100>
+if you can see this, you browser doesn't understand SANDBOX IFRAME.
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:306]:
```suggestion
if you can see this, you browser doesn't understand SANDBOX IFRAME.
```
```suggestion
if you can see this, your browser doesn't understand SANDBOX IFRAME.
```

Comment (carrrina) [inline web-basics-browser-security-model/README.md:306]:
```suggestion
if you can see this, you browser doesn't understand SANDBOX IFRAME.
```
```suggestion
if you can see this, your browser doesn't understand SANDBOX IFRAME.
```

File: web-basics-browser-security-model/README.md [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+</iframe>
+```
+
+Access the page via browser http://localhost.
+
+The value of the sandbox attribute can either be just sandbox (then all restrictions are applied), or a space-separated list of pre-defined values that will REMOVE the particular restrictions.
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:312]:
```suggestion
The value of the sandbox attribute can either be just sandbox (then all restrictions are applied), or a space-separated list of pre-defined values that will REMOVE the particular restrictions.
```
```suggestion
The value of the sandbox attribute can either be just sandbox (then all restrictions are applied), or a space-separated list of predefined values that will REMOVE the particular restrictions.
```

Comment (carrrina) [inline web-basics-browser-security-model/README.md:312]:
```suggestion
The value of the sandbox attribute can either be just sandbox (then all restrictions are applied), or a space-separated list of pre-defined values that will REMOVE the particular restrictions.
```
```suggestion
The value of the sandbox attribute can either be just sandbox (then all restrictions are applied), or a space-separated list of predefined values that will REMOVE the particular restrictions.
```

File: web-basics-browser-security-model/README.md [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+
+# Same-Origin Policy
+
+The same-origin policy is a web browser security mechanism that aims to prevent websites from attacking each other. The same-origin policy restricts scripts on one origin from accessing data from another origin.
+
+## Why we need Same-Origin Policy?
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:318]:
```suggestion
## Why we need Same-Origin Policy?
```
```suggestion
## Why do we need Same-Origin Policy?
```

Comment (carrrina) [inline web-basics-browser-security-model/README.md:318]:
```suggestion
## Why we need Same-Origin Policy?
```
```suggestion
## Why do we need Same-Origin Policy?
```

File: web-basics-browser-security-model/README.md [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+
+Postman is a tool through which you can send HTTP requests. It can automate tests and integrate them into CI/CD pipeline. You can simulate endpoints to test without having to set up a backend. It can also monitor certain endpoints to check their health and performance. [[6]](https://www.postman.com/)
+
+## HTTPie
+
+HTTPie is also used to send HTTP requests. It can be used in the terminal, instead of utilities such as **curl** and **wget**. It has an easy syntax, comes with highlighting and JSON assets. [[7]](https://httpie.io/)
+
+## Hoppscotch
+
+Hoppscotch is a tool that you can use from your browser or as a browser extension to quickly test certain endpoints. It also assetss MQTT or GraphQL requests. [[8]](https://hoppscotch.io/)
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:370]: **assetss**?

Comment (carrrina) [inline web-basics-browser-security-model/README.md:370]: **assetss**?

File: web-basics-browser-security-model/README.md [hunk 9]
```diff
@@ -0,0 +1,403 @@
+# Introduction
+
+[Session 01 slides](https://docs.google.com/presentation/d/19ntegcXmtk2DGBfr-FMBvEVTitvqaxaOROKlDUnNhEM/edit?usp=sharing)
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:3]: Can we also upload the slides in PDF format, in a `slides/` directory, as instructed [here](https://github.com/security-summer-school/template-public)?

Comment (carrrina) [inline web-basics-browser-security-model/README.md:3]: Can we also upload the slides in PDF format, in a `slides/` directory, as instructed [here](https://github.com/security-summer-school/template-public)?

File: web-basics-browser-security-model/README.md [hunk 10]
```diff
@@ -0,0 +1,403 @@
...
+- [5] https://portswigger.net/web-security/cors/same-origin-policy
+- [6] https://www.postman.com/
+- [7] https://httpie.io/
+- [8] https://hoppscotch.io/
+
+# Challenges
+
+**1.** The below image represents a snippet with DevTools containing information about a web application. What can you discover in the next image ? Is there any useful information from a security point of view ? Write the answer to the instructor.
+
+![inspect-element](https://drive.google.com/uc?export=view&id=17xxCWiNyMQtCec5tRYU-lNOk4vwr9EUV)
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:387]: Can we also upload this image here, in `assets/` or in `activities/inspect-element/public/`? And not depend on external websites anymore.

Comment (carrrina) [inline web-basics-browser-security-model/README.md:387]: Can we also upload this image here, in `assets/` or in `activities/inspect-element/public/`? And not depend on external websites anymore.

File: web-basics-browser-security-model/README.md [hunk 11]
```diff
@@ -0,0 +1,403 @@
...
+
+https://imgur.com/a/47j3r5Y
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:389]: Don't forget to delete this after.

Comment (carrrina) [inline web-basics-browser-security-model/README.md:389]: Don't forget to delete this after.

File: web-basics-browser-security-model/README.md [hunk 12]
```diff
@@ -0,0 +1,403 @@
...
+
+**2.** [Cockroach](https://sss-ctf.security.cs.pub.ro/challenges?category=web-sessions)
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:391]: Will these links be updated later? To point to the specific challenge?

Comment (carrrina) [inline web-basics-browser-security-model/README.md:391]: Will these links be updated later? To point to the specific challenge?

File: web-basics-browser-security-model/activities/cockroach/sol/solution [hunk 13]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+PORT=8080
+
+if [[ $1 == "local" ]]
+then
+    remote='http://127.0.0.1:'$PORT
```
Comment (carrrina) [inline web-basics-browser-security-model/activities/cockroach/sol/solution:6]: Again I'm having trouble with the variable name `remote`. A better one would be `url` or `address`

Comment (carrrina) [inline web-basics-browser-security-model/activities/cockroach/sol/solution:6]: Again I'm having trouble with the variable name `remote`. A better one would be `url` or `address`

Comment (carrrina) [inline web-basics-browser-security-model/activities/cockroach/sol/solution:6]: If you update it, be kind enough to do it for all challenges

Comment (carrrina) [inline web-basics-browser-security-model/activities/cockroach/sol/solution:6]: If you update it, be kind enough to do it for all challenges

File: web-basics-browser-security-model/activities/my-special-name/sol/solution [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+then
+    remote='http://127.0.0.1:'$PORT
+elif [[ $1 == "remote" ]] && [[ -z $2 ]] 
+then
+    remote='http://141.85.224.157:'$PORT
+else
+    remote=$1':'$2
+fi
+
+# Gimme
```
Comment (carrrina) [inline web-basics-browser-security-model/activities/my-special-name/sol/solution:14]: This challenge is My Special Name, not Gimme

Comment (carrrina) [inline web-basics-browser-security-model/activities/my-special-name/sol/solution:14]: This challenge is My Special Name, not Gimme

**Comment by carrrina** [comment]
Do we really not want to include the session number in the folder name anymore? :(

**Comment by carrrina** [comment]
Do we really not want to include the session number in the folder name anymore? :(

**Comment by carrrina** [comment]
It would be nice if, in the future, we also added a README for each challenge solution, besides the `solution` script.

**Comment by carrrina** [comment]
It would be nice if, in the future, we also added a README for each challenge solution, besides the `solution` script.

**Review by carrrina — CHANGES_REQUESTED** [review]

File: web-basics-browser-security-model/README.md [hunk 15]
```diff
@@ -0,0 +1,403 @@
...
+- [1] https://skorks.com/2010/05/what-every-developer-should-know-about-urls/
+- [2] https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+- [3] https://owasp.org/www-project-top-ten/
+- [4] https://github.com/denimgroup/denimgroup-vulnerability-examples
+- [5] https://portswigger.net/web-security/cors/same-origin-policy
+- [6] https://www.postman.com/
+- [7] https://httpie.io/
+- [8] https://hoppscotch.io/
+
+# Challenges
```
Comment (carrrina) [inline web-basics-browser-security-model/README.md:383]:
```suggestion
# Activities
```

**Comment by carrrina** [comment]
All the `solution` scripts should be named `solution.sh` as per [here](https://github.com/security-summer-school/template-internal)

**Comment by gabrielmocanu** [comment]
@carrrina  Where did I include the session number in folder name? As I see the folder name is web-basics-browser-security-model.

File: web-basics-browser-security-model/README.md (hunk 12 above)
Comment (gabrielmocanu) [reply web-basics-browser-security-model/README.md:391]: Yes. We will update after we have the domains on CyberEdu.

**Comment by gabrielmocanu** [comment]
I didn't add a README for each challenge because here the solutions are pretty straightforward.

File: web-basics-browser-security-model/README.md (hunk 8 above)
Comment (gabrielmocanu) [reply web-basics-browser-security-model/README.md:370]: I don't know what was here but I changed with:
"""
It also supports MQTT or GraphQL requests.
"""

File: web-basics-browser-security-model/README.md (hunk 11 above)
Comment (gabrielmocanu) [reply web-basics-browser-security-model/README.md:389]: Done.

File: web-basics-browser-security-model/README.md (hunk 10 above)
Comment (gabrielmocanu) [reply web-basics-browser-security-model/README.md:387]: Yes. I think it's the best solution.

File: web-basics-browser-security-model/activities/cockroach/sol/solution (hunk 13 above)
Comment (gabrielmocanu) [reply web-basics-browser-security-model/activities/cockroach/sol/solution:6]: I changed the naming for all challenges.

Comment (gabrielmocanu) [reply web-basics-browser-security-model/activities/cockroach/sol/solution:6]: Done.

File: web-basics-browser-security-model/activities/my-special-name/sol/solution (hunk 14 above)
Comment (gabrielmocanu) [reply web-basics-browser-security-model/activities/my-special-name/sol/solution:14]: Yeah, it seems that I didn't include the right script for my special name.
It should be ok now.

File: web-basics-browser-security-model/README.md (hunk 9 above)
Comment (gabrielmocanu) [reply web-basics-browser-security-model/README.md:3]: Yup. I did the markdown file for the slides but I didn't upload a pdf.
I think the markdown file it's enough for participants.

**Comment by gabrielmocanu** [comment]
@carrrina  I did all the changes that you suggested.
Do you can we can merge this?

**Review by carrrina — APPROVED** [review]

**Pull request merged**